src/components/Stakeholders/Add.js

```diff
@@ -0,0 +1,258 @@
+import React, { useState, useContext } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  Button,
+  Checkbox,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  Grid,
+  MenuItem,
+  FormControl,
+  FormControlLabel,
+  FormGroup,
+  TextField,
+} from '@material-ui/core';
+import GSInputLabel from '../common/InputLabel';
+import { StakeholdersContext } from 'context/StakeholdersContext';
+
+const useStyles = makeStyles({
+  root: {
+    '& .MuiTextField-root': {
+      margin: '0 5px',
+    },
+    width: '50%',
+  },
+});
+
+const initialState = {
+  type: 'Not set',
+  org_name: '',
+  first_name: '',
+  last_name: '',
+  email: '',
+  phone: '',
+  password: '',
+  salt: '',
+  offering_pay_to_plant: false,
+  website: '',
+  logo_url: '',
+  map: '',
+};
+
+function AddStakeholder() {
+  const classes = useStyles();
+  const { createStakeholder } = useContext(StakeholdersContext);
+  const [open, setOpen] = useState(false);
+  const [data, setData] = useState(initialState);
+
+  const defaultTypeList = [
+    {
+      name: 'Not set',
+      value: 'Not set', // undefined works
+    },
+    {
+      name: 'Organization',
+      value: 'Organization',
+    },
+    {
+      name: 'Grower',
+      value: 'Person',
+    },
+  ];
+
+  const openModal = () => {
+    setOpen(true);
+  };
+
+  const closeModal = () => {
+    setData(initialState);
+    setOpen(false);
+  };
+
+  const handleChange = (e) => {
+    setData({ ...data, [e.target.name]: e.target.value });
+  };
+
+  const submit = () => {
```

Comment:
When adding a Stakeholder, if a user leaves required entries unanswered and attempts to submit, a Validation Error is thrown in the backend. However, I would prevent a request from being sent, which can be done by the frontend. I see that you have passed `required` flags in some form fields, however I'd suggest looking at [this example](https://github.com/Greenstand/treetracker-admin-client/pull/499/files#diff-217d3caf78535283104fa78750472ee22e659c1f349586b09f61ab6b27191c3bR67) that seems to work, which uses a </form> component. This can help in reducing the number of requests sent to the API and prevent the user from making mistakes in the UI leading to incorrect entries in the DB.